TorSharp/Tools/LineByLineConfigurer.cs

```diff
@@ -57,14 +57,24 @@ public LineByLineConfigurer(IConfigurationDictionary configurationDictionary, IC
                     }
                 }
 
+                // If the original file exists, remove it before copying over the temporary file.
                 if (File.Exists(path))
                 {
-                    File.Replace(temporaryPath, path, null);
-                }
-                else
-                {
-                    File.Move(temporaryPath, path);
+                    string backupPath = path + ".bak";
+
+                    // If there has already been a backup, just delete the original.
+                    if (File.Exists(backupPath))
+                    {
+                        File.Delete(path);
```

Comment:
If this code is executed over and over, the same backup file stays forever. The SO answer that you mentioned continuously replaces the backup with the latest copy.

I think having a backup file is a good idea (especially considering that these backups don't build up -- there is only ever one per path), but could we perhaps do

```csharp
File.Delete(backupPath);
File.Replace(temporaryPath, path, backupPath);
```

If this works for you, I can merge your PR and make this revision (as well as releasing a new version of the NuGet package).